### PR TITLE
PYTHON-4541 Add automatic handling of python following version

### PR DESCRIPTION
**Pull request by blink1073**
Tested with https://github.com/blink1073/winkerberos/actions/runs/10134646984

### CODE REVIEW

File: python/publish/handle_following_version.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+version = Version(sys.argv[1])
+new_version = None
+
+if version.is_devrelease:
+    # For dev releases, increment the dev release number.
+    new_version = f"{version.major}.{version.minor}.{version.micro}.dev{version.dev + 1}"
+elif version.is_prerelease:
+    # For pre releases, go back to dev release.
+    rel_type, rel_num = version.pre
```
Comment (NoahStapp) [inline python/publish/handle_following_version.py:12]: Are these variables used?

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: python/publish/handle_following_version.py (hunk 1 above)
Comment (blink1073) [reply python/publish/handle_following_version.py:12]: Good catch!

Comment (NoahStapp) [reply python/publish/handle_following_version.py:12]: I don't see any changes pushed, should there be?

Comment (blink1073) [reply python/publish/handle_following_version.py:12]: lol I didn't save the file

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**